### PR TITLE
refactor(devtools): dataflow implementation of client

### PR DESCRIPTION
**Pull request by Asuka109**
## Summary

This pull request simplify initialization of RPC connection and client states. It will be seamless to the users.

Which merge those source files:

[client/src/entries/client/rpc/index.ts](https://github.com/web-infra-dev/modern.js/blob/c4948524b30accef768723a2a43be2f98f319f25/packages/devtools/client/src/entries/client/rpc/index.ts)
[client/src/entries/client/rpc/mount.ts](https://github.com/web-infra-dev/modern.js/blob/c4948524b30accef768723a2a43be2f98f319f25/packages/devtools/client/src/entries/client/rpc/mount.ts)
[client/src/entries/client/rpc/server.ts](https://github.com/web-infra-dev/modern.js/blob/c4948524b30accef768723a2a43be2f98f319f25/packages/devtools/client/src/entries/client/rpc/server.ts)
[client/src/entries/client/stores/index.tsx](https://github.com/web-infra-dev/modern.js/blob/c4948524b30accef768723a2a43be2f98f319f25/packages/devtools/client/src/entries/client/stores/index.tsx)
[client/src/entries/client/constants.tsx](https://github.com/web-infra-dev/modern.js/blob/c4948524b30accef768723a2a43be2f98f319f25/packages/devtools/client/src/entries/client/constants.tsx)

into a new file with a clearer data flow and implementation.

[client/src/entries/mount/state.ts](https://github.com/web-infra-dev/modern.js/blob/5e6c84dec636ed4f627dda5cf9ba1fd880c8ea77/packages/devtools/client/src/entries/mount/state.ts)

There are some changes to these files:

kit/src/channel.ts
client/src/utils/hooks.ts
client/src/utils/react-devtools.ts
client/src/entries/mount/state.ts

The remaining changes are basicly only made to align with the above.

## Checklist

- [ ] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/devtools/client/jest.config.js [hunk 1]
```diff
@@ -0,0 +1,198 @@
+/**
+ * For a detailed explanation regarding each configuration property, visit:
+ * https://jestjs.io/docs/configuration
```
Comment (chenjiahan) [inline packages/devtools/client/jest.config.js:3]: Too many unused comments in this file, should we preserve these comments?

File: packages/devtools/client/package.json [hunk 2]
```diff
@@ -71,6 +73,8 @@
     "type-fest": "^4.1.0",
     "typescript": "~5.0.4",
     "ufo": "^1.3.0",
-    "valtio": "^1.11.1"
+    "valtio": "^1.11.1",
+    "babel-jest": "^29.5.0",
```
Comment (chenjiahan) [inline packages/devtools/client/package.json:77]: why not @swc/jest

Comment (chenjiahan) [reply packages/devtools/client/package.json:77]: can we just extends @scripts/jest-config

File: packages/devtools/kit/tests/channel.test.ts [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+class MockPostMessageTarget implements PostMessageTarget {
+  private listeners: Record<string, PostMessageListener[]> = {};
+
+  postMessage(
+    message: any,
+    targetOrigin: string,
+    transfer?: Transferable[],
+  ): void {
+    // 模拟 postMessage 行为，立即触发 message 事件
```
Comment (chenjiahan) [inline packages/devtools/kit/tests/channel.test.ts:16]: English by default

File: packages/devtools/kit/src/channel.ts [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+  constructor() {
+    super('Connection to target message channel port timed out');
+  }
+}
+
+export type PostMessageListener = (ev: MessageEvent) => void;
+
+export interface PostMessageTarget {
+  postMessage: (
+    message: any,
```
Comment (chenjiahan) [inline packages/devtools/kit/src/channel.ts:71]: can we use unknown instead of any

Comment (chenjiahan) [reply packages/devtools/kit/src/channel.ts:71]: so many anys in this PR...

File: packages/devtools/client/package.json (hunk 2 above)
Comment (Asuka109) [reply packages/devtools/client/package.json:77]: I forgot about the shared configs. Let me rewrite it.

File: packages/devtools/kit/tests/channel.test.ts (hunk 3 above)
Comment (Asuka109) [reply packages/devtools/kit/tests/channel.test.ts:16]: Sure.

File: packages/devtools/kit/src/channel.ts (hunk 4 above)
Comment (Asuka109) [reply packages/devtools/kit/src/channel.ts:71]: It just implement the [`ChannelOptions`](https://github.com/antfu/birpc/blob/b82bef7dbc804525be7b4afca127997160cf9830/src/index.ts#L9) interface from `birpc` and won't be used by any other developers but just pass to the birpc instance. But I found the restrict typing won't make it unaccpetable to birpc. So I did it in the latest commit.

File: packages/devtools/client/package.json (hunk 2 above)
Comment (Asuka109) [reply packages/devtools/client/package.json]: Done~

File: packages/devtools/client/babel.config.js [hunk 5]
```diff
@@ -0,0 +1,6 @@
+module.exports = {
```
Comment (chenjiahan) [inline packages/devtools/client/babel.config.js:1]: can we delete babel.config.js now

**Review by chenjiahan — APPROVED** [review]

**Pull request merged**